spec/services/api/trainees/withdraw_response_spec.rb

```diff
@@ -28,8 +32,9 @@
     it "change the trainee" do
       expect {
         subject
-      } .to change { trainee.reload.withdraw_reasons_details }.from(nil).to(params[:withdraw_reasons_details])
+      } .to change { trainee.reload.current_withdrawal&.trigger }.from(nil).to("provider")
       .and change { trainee.reload.withdraw_date }.from(nil)
+      .and change { trainee.reload.current_withdrawal&.future_interest }.from(nil).to("no")
       .and change { trainee.reload.state }.from("trn_received").to("withdrawn")
```

Comment:
Unless I'm missing something else in the spec, it looks like the `reasons` are never checked to have changed on the trainee. It would be worth adding.